src/uwtools/tests/drivers/test_mpas.py

```diff
@@ -0,0 +1,234 @@
+    atm_cfg_src, sfc_cfg_src = [str(tmp_path / (x + ".in")) for x in [atm_cfg_dst, sfc_cfg_dst]]
+    config["mpas"].update({key: {atm_cfg_dst: atm_cfg_src, sfc_cfg_dst: sfc_cfg_src}})
+    path = tmp_path / "config.yaml"
+    with open(path, "w", encoding="utf-8") as f:
+        yaml.dump(config, f)
+    driverobj = mpas.MPAS(config=path, cycle=cycle, batch=True)
+    atm_dst, sfc_dst = [tmp_path / (x % cycle.strftime("%H")) for x in [atm, sfc]]
+    assert not any(dst.is_file() for dst in [atm_dst, sfc_dst])
+    atm_src, sfc_src = [Path(str(x) + ".in") for x in [atm_dst, sfc_dst]]
+    for src in (atm_src, sfc_src):
+        src.touch()
+    getattr(driverobj, task)()
+    assert all(getattr(dst, test)() for dst in [atm_dst, sfc_dst])
+
+
+def test_MPAS_namelist_file(driverobj):
+    dst = driverobj._rundir / "namelist.atmosphere"
+    # Path(driverobj._driver_config["namelist"]["base_file"]).touch()
+    assert not dst.is_file()
+    driverobj.namelist_file()
+    assert dst.is_file()
+    assert isinstance(f90nml.read(dst), f90nml.Namelist)
+
+
+def test_MPAS_namelist_missing(driverobj):
+    path = driverobj._rundir / "namelist.atmosphere"
+    del driverobj._driver_config["namelist"]
+    with raises(UWConfigError) as e:
+        assert driverobj.namelist_file()
+    assert str(e.value) == ("Provide either a 'namelist' YAML block or the %s file" % path)
+
+
+def test_MPAS_provisioned_run_directory(driverobj):
+    with patch.multiple(
+        driverobj,
+        boundary_files=D,
+        files_copied=D,
+        files_linked=D,
+        namelist_file=D,
+        runscript=D,
+        streams_file=D,
+    ) as mocks:
+        driverobj.provisioned_run_directory()
+    for m in mocks:
+        mocks[m].assert_called_once_with()
+
+
+def test_MPAS_run_batch(driverobj):
+    with patch.object(driverobj, "_run_via_batch_submission") as func:
+        driverobj.run()
+    func.assert_called_once_with()
+
+
+def test_MPAS_run_local(driverobj):
+    driverobj._batch = False
+    with patch.object(driverobj, "_run_via_local_execution") as func:
+        driverobj.run()
+    func.assert_called_once_with()
+
+
+def test_MPAS_runscript(driverobj):
+    dst = driverobj._rundir / "runscript.mpas"
+    assert not dst.is_file()
+    driverobj._driver_config["execution"].update(
+        {
+            "batchargs": {"walltime": "00:01:00"},
+            "envcmds": ["cmd1", "cmd2"],
+            "mpicmd": "runit",
+            "threads": 8,
+        }
+    )
+    driverobj._config["platform"] = {"account": "me", "scheduler": "slurm"}
+    driverobj.runscript()
+    with open(dst, "r", encoding="utf-8") as f:
+        lines = f.read().split("\n")
+    # Check directives:
+    assert "#SBATCH --account=me" in lines
+    assert "#SBATCH --time=00:01:00" in lines
+    # Check environment commands:
+    assert "cmd1" in lines
+    assert "cmd2" in lines
+    # Check execution:
+    assert "test $? -eq 0 && touch %s/done" % driverobj._rundir
```

Comment:
IMO, for methods that have been factored out to `Driver` -- either just with the bulk of their logic moved, or the entire method moved there -- it's enough in these per-driver unit tests to mock that `Driver` method or function and then `assert` here that it was called with the expected arguments. Then, we can concentrate our efforts on vigorously testing those core, shared method/functions in the `Driver` unit tests, and all drivers will benefit, without duplication of effort. Here, I'm not sure if the tests of `runscript()`, `driver_config()`, or `resources()` add much over what's already tested in the `Driver` tests. Please let me know if I'm wrong!

Reply:
Without `test_MPAS_runscript()` we don't have 100% test coverage. The `driver_config()` and `resources()` tests do seem unnecessary, though.

Reply:
I didn't mean that this necessarily has to change now. Just that it might not have been necessary to write quite so much test code, given that the MPAS drivers call `Driver` in cases that are already well-tested. Duplicating similar tests across test modules does mean that more code may have to be updated if something changes that effects all of them.

Reply:
You do need a method to exercise it, but it can be simpler. See the `runscript()` tests in the other drivers.